src/shared/mount-util.c

```diff
@@ -1858,11 +1858,20 @@ int path_is_network_fs_harder(const char *path) {
 
         assert(path);
 
+        if (!path_is_absolute(path))
+                return -EINVAL;
+
         ret = path_is_network_fs(path);
+        if (ret == -ENOENT)
+                return ret;
         if (ret > 0)
                 return true;
 
         r = path_get_mount_info(path, &fstype, &options);
+        if (r == -EINVAL) {
+                log_debug_errno(r, "Failed to find mount point '%s', maybe not a mount point, assuming not a network filesystem: %m", path);
```

Comment:
so i think we should tweak path_get_mount_info():

1. it should get the mount id from the path via path_get_mnt_id()
2. then search through /proc/self/mountinfo via the mount id. libmount currently doesn't provide a direct api for this (i asked for it in https://githuhttps://github.com/util-linux/util-linux/issues/3275b.com/util-linux/util-linux/issues/3275), but we can just iterate through the table ourselves until we find an entry where mnt_fs_get_id() checks out. (a new api in libmount for this wouldn't algorithmically do anything better anyway, at least for now.)

benefit of this: this code would work on any inode, not just mount point inodes.